### PR TITLE
Set up prow for release 1.6

### PR DESCRIPTION
**Pull request by brian-avery**

### CODE REVIEW

File: prow/config/jobs/api-1.6.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+branches:
+- release-1.6
+image: gcr.io/istio-testing/build-tools:master-latest
```
Comment (howardjohn) [inline prow/config/jobs/api-1.6.yaml]: this will change immediately after right? since we should use a pinned version normally

**Review by howardjohn — APPROVED** [review]
lgtm!

**Comment by clarketm** [comment]
> /test pull-test-infra-check-testgrid-config

@brian-avery to fix this, add the new dashboard names (e.g. `istio_release-1.6_istio.io_periodic`) to https://github.com/istio/test-infra/blob/master/testgrid/default.yaml. You can use the 1.5 ones as a reference (they are most likely the same if the jobs are the same).

File: prow/config/jobs/istio-1.6.yaml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+- command:
+  - ../test-infra/tools/automator/automator.sh
+  - --org=istio
+  - --repo=istio.io
+  - --token-path=/etc/github-token/oauth
+  - --cmd=make update_ref_docs
+  - --dry-run
+  modifiers:
+  - optional
+  name: update-ref-docs-dry-run
```
Comment (clarketm) [inline prow/config/jobs/istio-1.6.yaml]: We only build ref docs from `master`, so I think this job can be omitted (which will also resolve the test failure w/ testgrid).

**Review by clarketm — APPROVED** [review]

File: prow/config/jobs/istio.io-1.6.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- command:
+  - entrypoint
+  - prow/integ-suite-kind.sh
+  - test.kube.postsubmit
+  name: k8s-tests
+  requirements:
+  - kind
+  - gcp
+  type: postsubmit
+- command:
```
Comment (clarketm) [inline prow/config/jobs/istio.io-1.6.yaml]: Remove this job as well.  It is only needed on `master`. ref:  https://github.com/istio/test-infra/issues/1626

**Pull request merged**